test/rule-syntax-check.py

```diff
@@ -28,10 +28,10 @@
 if not rules_files:
     sys.exit('Specify files to test as arguments')
 
-no_args_tests = re.compile(r'(ACTION|DEVPATH|KERNELS?|NAME|SYMLINK|SUBSYSTEMS?|DRIVERS?|TAG|PROGRAM|RESULT|TEST)\s*(?:=|!)=\s*"([^"]*)"$')
-args_tests = re.compile(r'(ATTRS?|ENV|TEST){([a-zA-Z0-9/_.*%-]+)}\s*(?:=|!)=\s*"([^"]*)"$')
-no_args_assign = re.compile(r'(NAME|SYMLINK|OWNER|GROUP|MODE|TAG|RUN|LABEL|GOTO|OPTIONS|IMPORT)\s*(?:\+=|:=|=)\s*"([^"]*)"$')
-args_assign = re.compile(r'(ATTR|ENV|IMPORT|RUN){([a-zA-Z0-9/_.*%-]+)}\s*(=|\+=)\s*"([^"]*)"$')
+no_args_tests = re.compile(r'(ACTION|DEVPATH|KERNELS?|NAME|SYMLINK|SUBSYSTEMS?|DRIVERS?|TAG|PROGRAM|RESULT|TEST)\s*(?:=|!)=\s*"(.*)"$')
+args_tests = re.compile(r'(ATTRS?|ENV|TEST){([a-zA-Z0-9/_.*%-]+)}\s*(?:=|!)=\s*"(.*)"$')
+no_args_assign = re.compile(r'(NAME|SYMLINK|OWNER|GROUP|MODE|TAG|RUN|LABEL|GOTO|OPTIONS|IMPORT)\s*(?:\+=|:=|=)\s*"(.*)"$')
+args_assign = re.compile(r'(ATTR|ENV|IMPORT|RUN){([a-zA-Z0-9/_.*%-]+)}\s*(=|\+=)\s*"(.*)"$')
```

Comment:
Hmm, `.` is greedy, so this doesn't really work: most of the line can be matched to a single key=val assignment.
```pycon
>>> import re
>>> re.match(r'abc="(.*)"', 'abc="abc" def="def"').group(1)
'abc" def="def'
```

Reply:
Trying to rewrite the regular expressions, I realized that I don't fully understand how `udev` parses rules. Anyway, I was able to come up with `"(?:[^\\"]|\\"|\\(?!"))*"$`, which I think should be close enough to what `udev` does.

I would also change all `(`s to `(?:`s, but that's nitpicking.

Reply:
`split(',')` in the middle of the script doesn't seem to be correct, because in practice commas are optional and can be omitted. As a result, the first commit appears to fix a false positive.

I'm also wondering if all the missing rules including `70-uaccess.rules` should be added to `all_rules` so that `rule-syntax-check` will check them automatically too.

Reply:
@keszybz you're right, I forgot that ',' was optional which doesn't seem to be a good idea but it's too late.

Reply:
@evverx @keszybz I'm wondering if we shouldn't introduce a new udev tool which would reuse the actual rules parser of udev instead of relying on a script that tries to mimic the behavior of udev...

And other rules providers could use it.

Reply:
@fbuihuu, a tool using the actual parser sounds great to me. It would really tell us if the rules are correct. On the other hand, I've come to realize that a script getting people to use commas (which I think make rules more readable) isn't as bad as I thought. Probably, it would make sense to have both and see them as a compiler and a linter.

Reply:
> On the other hand, I've come to realize that a script getting people to use commas (which I think make rules more readable) isn't as bad as I thought.

Agreed.

> Probably, it would make sense to have both and see them as a compiler and a linter.

Hmm I don't see the point to have both. But until something else is created we need to fix the rules of the parser checker.

Since you found a correct fix, could you submit ?

Reply:
@fbuihuu, I could, but, to be honest, my ability to send PRs is a bit limited now (I'm having difficulties getting a 2FA code), so it would be much easier and quicker if you could just update this PR.